### PR TITLE
[Instrumentation.Quartz] Nullable

### PR DESCRIPTION
**Pull request by Kielek**
Towards #894
## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* ~~[ ] Appropriate `CHANGELOG.md` updated for non-trivial changes~~
* ~~[ ] Design discussion issue #~~
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by Kielek** [comment]
@maldago, could you please review?

**Review by maldago — APPROVED** [review]
I don't see any issue with this. Thanks @Kielek

File: src/OpenTelemetry.Instrumentation.Quartz/QuartzInstrumentationOptions.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public class QuartzInstrumentationOptions
     /// <para>object: the raw object from which additional information can be extracted to enrich the activity.
     /// The type of this object depends on the event, which is given by the above parameter.</para>
     /// </remarks>
-    public Action<Activity, string, object> Enrich { get; set; }
+    public Action<Activity, string, object?>? Enrich { get; set; }
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Quartz/QuartzInstrumentationOptions.cs:46]: I think the better thing to do would be to not mark `object` nullable and update the code to call `Enrich` only when the `object` passed to it is not null.

**Comment by Kielek** [comment]
@maldago, could you please also check #1201?

(sorry for mixing topics, but I do not found any better way to send you a message.

File: src/OpenTelemetry.Instrumentation.Quartz/QuartzInstrumentationOptions.cs (hunk 1 above)
Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.Quartz/QuartzInstrumentationOptions.cs]: Please check 8b32c4b3837e37eeb2bfef7edc5e16d414a58e57

Comment (maldago) [reply src/OpenTelemetry.Instrumentation.Quartz/QuartzInstrumentationOptions.cs]: I think the update is fair.

**Pull request merged**